src/Message/AbstractRequest.php

```diff
@@ -30,26 +28,15 @@ public function getTransactionId()
 
     protected function sendRequest($method, $endpoint, $data = null)
     {
-        $this->httpClient->getEventDispatcher()->addListener('request.error', function (Event $event) {
-            /**
-             * @var \Guzzle\Http\Message\Response $response
-             */
-            $response = $event['response'];
-
-            if ($response->isError()) {
-                $event->stopPropagation();
-            }
-        });
-
         $httpRequest = $this->httpClient->createRequest(
             $method,
             $this->endpoint . $endpoint,
             array(
                 'Authorization' => 'Bearer ' . $this->getApiKey()
             ),
-            $data
+            json_encode($data)
```

Comment:
Looking at https://github.com/mollie/mollie-api-php/blob/74fd1a03a8841d9306fc2cf0fae3a887f291aaba/src/Mollie/API/Resource/Base.php#L253-L273 I would think you are right btw with json_encode, just not sure why it worked before.